test/unit/proxy_api/proxy_exception_test.rb

```diff
@@ -0,0 +1,21 @@
+require 'test_helper'
+
+class ProxyApiExceptionTest < ActiveSupport::TestCase
+  test 'exception response body is added if available' do
+    fake_exception = OpenStruct.new(:response => 'useful message from proxy',
+                                    :code => 400)
+    proxy_exception = ProxyAPI::ProxyException.new(
+      'fakeurl',
+      OpenStruct.new(:message => 'some error'),
+      'unable to do something')
+    proxy_exception.expects(:exception).returns(fake_exception)
+    assert_match(/useful message from proxy/, proxy_exception.message)
```

Comment:
It'd be useful if this could test the entire message - or more of the concatenation, not only the proxy_exception_response bit.